mindmeld/components/dialogue.py

```diff
@@ -1011,6 +1026,38 @@ async def call_async(self, request, responder):
         """
         self(request, responder)
 
+    def invoke(self, request, responder):
+        """
+        Invoke slot-filling as a direct call without requiring a decorator.
+        """
+        self._invoked = True
+        kwargs = {'domain': request.domain, 'intent': request.intent}
+
+        # Set dialogue rule to auto_fill and
+        # store current rule for resetting once invoke is complete
+
+        name = self._entrance_handler.__name__
+
+        try:
+            _called_from = sys._getframe().f_back.f_code.co_name
+            self._previous_rule = self._app.app_manager.dialogue_manager.handler_map[_called_from]
+        except (KeyError, AttributeError):
+            return
+
+        try:
+            self._app.app_manager.dialogue_manager.add_dialogue_rule(name, self.__call__, **kwargs)
```

Comment:
Can we remove this rule after slot filing is done?

Reply:
Yup, that's exactly what the 916-922 block in ``_end_slot_fill`` does. If we don't remove it, the actions can't be repeated later :)

Reply:
Simplified the logic a bit, now it doesn't remove the rule anymore. It now sets it only facing the secondary function (and now the rule is also targeted), so this will stay in the set of rules but not hinder with any other rule in the app. I don't think there is any need to explicitly get rid of it.